### PR TITLE
Timezone returns wrong value during DST

### PR DESCRIPTION
**Pull request by DConmy**
https://github.com/emscripten-core/emscripten/issues/11572

### CODE REVIEW

**Comment by tgriff34** [comment]
@kripken updated the test and it passes locally, not sure what the other failures are from though...mind if we get a review as well?

File: src/library.js [hunk 1]
```diff
@@ -998,17 +998,24 @@ LibraryManager.library = {
     if (_tzset.called) return;
     _tzset.called = true;
 
+    var currentYear = new Date().getFullYear();
+    var winter = new Date(currentYear, 0, 1);
+    var summer = new Date(currentYear, 6, 1);
+    var winterOffset = winter.getTimezoneOffset();
+    var summerOffset = summer.getTimezoneOffset();
+
+    function stdTimezoneOffset() {
```
Comment (kripken) [inline src/library.js:1007]: I don't know much about timezones, sorry. Is it obvious what this means? Please add a  comment explaining the name of the function + why the max of the winter and summer offsets is the value we need (how it relates to UTC etc.).

File: src/library.js [hunk 2]
```diff
@@ -998,17 +998,24 @@ LibraryManager.library = {
...
+        return Math.max(winterOffset, summerOffset);
+    }
```
Comment (kripken) [inline src/library.js:1009]: Also, why is this a function, and not just this?
```javascript
var stdTimezoneOffset = Max.max(..
```

**Review by kripken — COMMENTED** [review]
Thanks for the PR!

About the CI errors, I think they are because this branch needs to merge in the latest code (i.e., right now it's running older emscripten against the newer LLVM etc. that CI installs).

File: src/library.js (hunk 1 above)
Comment (tgriff34) [reply src/library.js]: Added a comment which hopefully clears this up, the comment below also goes into some detail as to why we need it and links to the pubs page.  I also merged with master so hopefully that fixes the CI issues. Thanks.

File: src/library.js [hunk 3]
```diff
@@ -980,17 +980,23 @@ LibraryManager.library = {
     if (_tzset.called) return;
     _tzset.called = true;
 
+    var currentYear = new Date().getFullYear();
+    var winter = new Date(currentYear, 0, 1);
+    var summer = new Date(currentYear, 6, 1);
+    var winterOffset = winter.getTimezoneOffset();
+    var summerOffset = summer.getTimezoneOffset();
+    // Local standard timezone offset. Local standard time is not adjusted for daylight savings.
+    // Using Math.max we can get retrieve the local standard time for the timezone.
```
Comment (kripken) [inline src/library.js:989]:
```suggestion
    // Daylight savings adjusts the time to a lower value, so take the maximum here.
```
Is this comment right? I am inferring this from you doing the maximum, I don't actually understand time zones...

Comment (tgriff34) [reply src/library.js]: Yup you're right, changed it hopefully it's a little more clear now.

**Review by kripken — APPROVED** [review]
Thanks, looks good now!

**Pull request merged**